packages/generator-sprotty/src/index.ts

```diff
@@ -0,0 +1,125 @@
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import Generator from 'yeoman-generator';
+import chalk from 'chalk';
+import path from 'path';
+
+const TEMPLATE_DIR = '../sprotty-local-template';
+const USER_DIR = '.';
+
+const PROJECT_NAME = /<%= project-name %>/g;
+const HTML_ELEMENT_ID = /<%= html-element-id %>/g;
+const OUT_PATH = /<%= out-path %>/g;
+
+const PROJECT_PATH = /<%= project-path %>/g;
+
+interface Answers {
+    projectName: string;
+    mainElementId: string;
+    generateStatic: boolean;
+}
+
+function description(...d: string[]): string {
+    return chalk.reset(chalk.dim(d.join(' ') + '\n')) + chalk.blueBright('?');
+}
+
+class SprottyGenerator extends Generator {
+    private answers: Answers;
+
+    writing(): void {
+        this.sourceRoot(path.join(__dirname, TEMPLATE_DIR));
+
+        for (const path of ['package.json', 'tsconfig.json', 'src']) {
+            this.fs.copy(
+                this.templatePath(path),
+                this._projectPath(path),
+                {
+                    process: content =>
+                        this._replaceTemplateWords(content),
+                    processDestinationPath: path =>
+                        this._replaceTemplateNames(path),
+                }
+            );
+        }
+        console.log('generate static: ' + this.answers.generateStatic)
+        if(this.answers.generateStatic) {
+            this.fs.copy(
+                this.templatePath('static'),
+                this._projectPath('static'),
+                {
+                    process: content =>
+                        this._replaceTemplateWords(content),
+                    processDestinationPath: path =>
+                        this._replaceTemplateNames(path),
+                }
+            );
+        }
+    }
+
+    install(): void {
+        const extensionPath = this._projectPath();
+
+        const opts = { cwd: extensionPath };
+        this.spawnCommandSync('npm', ['install'], opts);
+        this.spawnCommandSync('npm', ['run', 'build'], opts);
+    }
+
+    async prompting(): Promise<void> {
+        this.answers = await this.prompt([
+            {
+                type: 'input',
+                name: 'projectName',
+                prefix: description(
+                    'Welcome to Sprotty!',
+                    'This tool generates a new Sprotty Project.\n',
+                    'The project name identifies the npm package and can be used by other packages to depend on this project.'
+                ),
+                message: 'Your project name:',
+                default: 'hello-world',
+            },
+            {
+                type: 'input',
+                name: 'mainElementId',
+                message: 'ID of your main html element for the sprotty diagram:',
```

Comment:
Proposal: Shorten the message to 'Main HTML element id' and add a `prefix` with more explanations, e.g. 'Your Sprotty diagram will be rendered inside this element.'